lib/mpl_toolkits/mplot3d/proj3d.py

```diff
@@ -171,23 +171,42 @@ def _ortho_transformation(zfront, zback):
 
 
 def _proj_transform_vec(vec, M):
-    vecw = np.dot(M, vec)
+    vecw = np.dot(M, vec.data)
     w = vecw[3]
     txs, tys, tzs = vecw[0]/w, vecw[1]/w, vecw[2]/w
+    if np.ma.isMA(vec[0]):  # we check each to protect for scalars
+        txs = np.ma.array(txs, mask=vec[0].mask)
+    if np.ma.isMA(vec[1]):
+        tys = np.ma.array(tys, mask=vec[1].mask)
+    if np.ma.isMA(vec[2]):
+        tzs = np.ma.array(tzs, mask=vec[2].mask)
     return txs, tys, tzs
 
 
 def _proj_transform_vec_clip(vec, M, focal_length):
-    vecw = np.dot(M, vec)
+    vecw = np.dot(M, vec.data)
     w = vecw[3]
-    txs, tys, tzs = vecw[0] / w, vecw[1] / w, vecw[2] / w
+    txs = np.ma.array(vecw[0] / w)
+    tys = np.ma.array(vecw[1] / w)
+    tzs = np.ma.array(vecw[2] / w)
     if np.isinf(focal_length):  # don't clip orthographic projection
         tis = np.ones(txs.shape, dtype=bool)
     else:
         tis = (-1 <= txs) & (txs <= 1) & (-1 <= tys) & (tys <= 1) & (tzs <= 0)
-    txs = np.ma.masked_array(txs, ~tis)
-    tys = np.ma.masked_array(tys, ~tis)
-    tzs = np.ma.masked_array(tzs, ~tis)
+    tis = np.ma.array(tis)
```

Comment:
Since `txs`, `tys`, and `tzs` are masked arrays, this should already be a masked array for the `else` case; you only need to change the true side to be `np.ma.ones` instead.